diff --git a/awesome.c b/awesome.c
index c71699d9..6abadff5 100644
--- a/awesome.c
+++ b/awesome.c
@@ -68,6 +68,10 @@ static struct timeval last_wakeup;
 
 /** current limit for the main loop's runtime */
 static float main_loop_iteration_limit = 0.1;
+/* Single trigger to only check the main_loop_iteration_limit after the WM is
+ * booted*/
+static char main_loop_iteration_booted = 0;
+static char main_loop_iteration_booted_ignore_time = 10;
 
 /** A pipe that is used to asynchronously handle SIGCHLD */
 static int sigchld_pipe[2];
@@ -466,10 +470,11 @@ a_glib_poll(GPollFD *ufds, guint nfsd, gint timeout)
     gettimeofday(&now, NULL);
     timersub(&now, &last_wakeup, &length_time);
     length = length_time.tv_sec + length_time.tv_usec * 1.0f / 1e6;
-    if (length > main_loop_iteration_limit) {
-        warn("Last main loop iteration took %.6f seconds! Increasing limit for "
-                "this warning to that value.", length);
+    if (length > main_loop_iteration_limit && main_loop_iteration_booted > main_loop_iteration_booted_ignore_time) {
+        warn("Last main loop iteration took %.6f seconds!", length);
         main_loop_iteration_limit = length;
+    } else if(main_loop_iteration_booted <= main_loop_iteration_booted_ignore_time) {
+        main_loop_iteration_booted++; 
     }
 
     /* Actually do the polling, record time of wakeup and check for new xcb events */
