diff --git a/lib/menubar/utils.lua b/lib/menubar/utils.lua
index b112397a..df510fa7 100644
--- a/lib/menubar/utils.lua
+++ b/lib/menubar/utils.lua
@@ -126,7 +126,16 @@ local all_icon_sizes = {
     '32x32',
     '24x24',
     '22x22',
-    '16x16'
+    '16x16',
+    '16',
+    '22',
+    '24',
+    '32',
+    '48',
+    '64',
+    '72',
+    '96',
+    '128'
 }
 
 --- List of supported icon exts.
@@ -192,6 +201,35 @@ local function get_icon_lookup_path()
             table.insert(app_in_theme_paths,
                          glib.build_filenamev({ icon_theme_directory,
                                                 size, 'categories' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'actions' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'devices' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'emblems' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'emotes' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'mimetypes' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'places' }))
+
+            table.insert(app_in_theme_paths,
+                         glib.build_filenamev({ icon_theme_directory,
+                                                size, 'status' }))
+
         end
     end
     add_if_readable(icon_lookup_path, app_in_theme_paths)
